jsonrpc/websocket.go

```diff
@@ -54,9 +54,19 @@ func (ws *Websocket) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	// TODO include connection information, such as the remote address, in the logs.
 
-	wsc := newWebsocketConn(conn, ws.rpc, ws.connParams, ws.listener)
+	wsc := newWebsocketConn(r.Context(), conn, ws.connParams, ws.listener)
 
-	err = wsc.ReadWriteLoop(r.Context())
+	for {
+		// Handle will read from the connection once.
+		var resp []byte
+		resp, err = ws.rpc.HandleReader(wsc.ctx, wsc)
+		if err != nil {
+			break
+		}
+		if _, err = wsc.Write(resp); err != nil {
+			break
+		}
+	}
```

Comment:
this loop will come back to bite us.

`HandleReader` uses a buffered reader, so it might over-read from the connection. Meaning that, while decoding a request we might consume a part of the second request and at the second iteration of the loop fail to decode the second request since the remaining bytes in the reader is now not a valid request.

Reply:
But this issue is not introduced in this PR, so lets just create an issue but we should fix it before fully rolling out WS